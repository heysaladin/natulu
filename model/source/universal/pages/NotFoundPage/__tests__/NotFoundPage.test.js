/* @flow */

import React from 'react'
import { Router } from 'react-router-dom'
import createMemoryHistory from 'history/createMemoryHistory'
import renderer from 'react-test-renderer'
import { mount, shallow, render } from 'enzyme'

import { NotFoundPage } from 'universal/pages/NotFoundPage/NotFoundPage'

test('NotFoundPage - snapshot test', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <Router history={history}>
      <NotFoundPage {...props} />
    </Router>
  )

  const tree = renderer.create(element).toJSON()

  expect(tree).toMatchSnapshot()
})

test('NotFoundPage - enzyme shallow render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <Router history={history}>
      <NotFoundPage {...props} />
    </Router>
  )

  const wrapper = shallow(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('NotFoundPage - enzyme full render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <Router history={history}>
      <NotFoundPage {...props} />
    </Router>
  )

  const wrapper = mount(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('NotFoundPage - enzyme static render', () => {
  const history = createMemoryHistory()

  const props = {}

  const element = (
    <Router history={history}>
      <NotFoundPage {...props} />
    </Router>
  )

  const wrapper = render(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})
