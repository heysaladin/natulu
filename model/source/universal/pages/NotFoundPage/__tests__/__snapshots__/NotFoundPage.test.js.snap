// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage - snapshot test 1`] = `
<section
  className="Page-vq5r1s-0 fmekfX"
>
  <section
    className="Center-s1p4naw5-0 dBuSNS"
  >
    <span
      className="Text-s1rrlk1p-0 kIrymH"
      fontSize="10vw"
    >
      Not Found Page
    </span>
  </section>
  <section
    className="Center-s1p4naw5-0 dBuSNS"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="Text-s1rrlk1p-0 cdvcYd"
        fontSize="5vw"
      >
        Go to Index Page
      </span>
    </a>
  </section>
</section>
`;
